### DISCUSSION TITLE
Return promise from SIWEConfig.createMessage

### DISCUSSION THREAD

**Discussion by anukul** [family/connectkit#323] [Ideas]
Update [`createMessage`](https://docs.family.co/connectkit/auth-with-custom-backend#SIWEConfig-createMessage) to return a promise, allowing users to fetch SIWE message from API, for use cases such as dynamically generating resources field etc.

**Comment by anukul** [comment]
We have done this in a fork - https://github.com/flood-protocol/connectkit/commit/0956e45, happy to make a PR

**Reply by lochie** [reply]
Sounds reasonable, not sure if this will have any side effects though but welcome to make a PR and I'll test it out.

**Reply by anukul** [reply]
https://github.com/family/connectkit/pull/324
